mvlearn/decomposition/multiviewica.py

```diff
@@ -0,0 +1,344 @@
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+#  this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Modified from source package https://github.com/hugorichard/multiviewica
+
+import numpy as np
+from scipy import linalg
+from sklearn.decomposition import PCA
+from multiviewica import multiviewica
+
+from .base import BaseDecomposer
+from ..preprocessing.repeat import ViewTransformer
+
+
+class MultiviewICA(BaseDecomposer):
+    r"""
+    Multiview ICA for which views share a common source but separate mixing
+    matrices.
+
+    Parameters
+    ----------
+    n_components : int, optional
+        Number of components to extract. If None, n_components is set to
+        the minimum number of features in the dataset.
+
+    noise : float, default=1.0
+        Gaussian noise level
+
+    max_iter : int, default=1000
+        Maximum number of iterations to perform
+
+    init : {'permica', 'groupica'} or np array of shape
+        (n_groups, n_components, n_components), default='permica'
+        If permica: initialize with perm ICA, if groupica, initialize with
+        group ica. Else, use the provided array to initialize.
+
+    preproc: 'pca' or a ViewTransformer-like instance,
+        default='pca'
+        Preprocessing method to use to reduce data.
+        If "pca", performs PCA separately on each view to reduce dimension
+        of each view.
+        Otherwise the dimension reduction is performed using the transform
+        method of the ViewTransformer-like object. This instance also needs
+        an inverse transform method to recover original data from reduced data.
```

Comment:
what do you think would be a good alternative as pca for a custom ViewTransformer?

Reply:
SRM for example

Reply:
Also people might want to use transformers that handle data that cannot hold in memory (this is quite common in fMRI)

Reply:
But then people can just use a custom preprocessing and use `make_pipeline(preprocessing, multiviewica)`, right?

Reply:
Yes that is right

Reply:
I like this ! it simplies the API